rinja_parser/src/expr.rs

```diff
@@ -685,12 +724,59 @@ impl<'a> Suffix<'a> {
     }
 
     fn call(i: &mut &'a str, level: Level<'_>) -> ParseResult<'a, Self> {
-        (move |i: &mut _| Expr::arguments(i, level, false))
-            .map(Self::Call)
+        (opt(|i: &mut _| call_generics(i, level)), |i: &mut _| {
+            Expr::arguments(i, level, false)
+        })
+            .map(|(generics, args)| Self::Call {
+                args,
+                generics: generics.unwrap_or_default(),
+            })
             .parse_next(i)
     }
 
     fn r#try(i: &mut &'a str) -> ParseResult<'a, Self> {
         preceded(skip_ws0, '?').map(|_| Self::Try).parse_next(i)
     }
 }
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct TyGenerics<'a> {
+    pub refs: usize,
+    pub ty: &'a str,
+    pub generics: Vec<WithSpan<'a, TyGenerics<'a>>>,
+}
+
+impl<'i> TyGenerics<'i> {
+    fn parse(i: &mut &'i str, level: Level<'_>) -> ParseResult<'i, WithSpan<'i, Self>> {
+        let start = *i;
+        (
+            ws(identifier_with_refs),
```

Comment:
This allows e.g. `&&u32`, but would fail for `&&core::primitive::u32`. I think we need to support paths in here. What do you think.

Reply:
Didn't even think about it but I agree.